src/shogun/lib/parameter_observers/ParameterObserverCV.cpp

```diff
@@ -74,9 +71,10 @@ void CParameterObserverCV::on_next(const shogun::TimedObservedValue& value)
 	}
 	else
 	{
-		SG_SERROR(
-		    "ParameterObserverCV: The observed value received is not of "
-		    "type CrossValidationStorage\n");
+		SG_SPRINT(
```

Comment:
why print and not error? also why not `REQUIRE`?

Reply:
It is because we do not want to make the application fail if the received value is not of the requested type. Let's imagine that I have several parameter observers attached to the same machine, let's call them `x` and `y`. They may receive the same emitted values from the machine, let's say A and B (with a different type). However, `x` may care only about A and `y` may care only about B. If `x` receives B, I do not want to make it crash the entire application. It just have to ignore that emitted value an process the next one.
See what I mean?

I can elaborate further in IRC if needed.